# HG changeset patch
# User Mike Hommey <mh+mozilla@glandium.org>
# Date 1635373828 0
# Node ID e819776339de975959fa508ccb7af50153f1f1bd
# Parent  6d9a180e160ffde67ae21d8dfb350f38012745be
Bug 1736830 - Be more consistent wrt package_desc in pkg_check_modules. r=firefox-build-system-reviewers,andi

Make @depends functions and direct values have the same level of
convenience.

Differential Revision: https://phabricator.services.mozilla.com/D129589

diff --git a/build/moz.configure/pkg.configure b/build/moz.configure/pkg.configure
--- a/build/moz.configure/pkg.configure
+++ b/build/moz.configure/pkg.configure
@@ -67,19 +67,27 @@ add_old_configure_assignment("PKG_CONFIG
 # - `when` a depends function that will determine whether to perform
 #   any checks (default is to always perform checks).
 # - `allow_missing` If set, failure to fulfill the package description
 #   will not result in an error or logged message, and any error message
 #   will be returned to the caller.
 #   Returns `True` when the package description is fulfilled.
 @template
 def pkg_check_modules(var, package_desc, when=always, allow_missing=False, config=True):
-    if isinstance(package_desc, (tuple, list)):
-        package_desc = " ".join(package_desc)
-    package_desc = dependable(package_desc)
+    @depends(dependable(package_desc), when=when)
+    def package_desc(desc):
+        if isinstance(desc, str):
+            desc = [desc]
+        if not isinstance(desc, (tuple, list)):
+            configure_error(
+                "package_desc must be a string or a tuple or list of strings"
+            )
+
+        return " ".join(desc)
+
     allow_missing = dependable(allow_missing)
 
     @depends(when, "--enable-compile-environment")
     def when_and_compile_environment(when, compile_environment):
         return when and compile_environment
 
     @depends(pkg_config, pkg_config_version, when=when_and_compile_environment)
     def check_pkg_config(pkg_config, version):
diff --git a/toolkit/moz.configure b/toolkit/moz.configure
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -1189,17 +1189,17 @@ def x11_libs(webrtc):
         # manually, ensure they're available.
         libs += [
             "xcomposite",
             "xcursor",
             "xdamage",
             "xfixes",
             "xi",
         ]
-    return " ".join(libs)
+    return libs
 
 
 pkg_check_modules("MOZ_X11", x11_libs, when=toolkit_gtk)
 
 
 # ASan Reporter Addon
 # ==============================================================
 option(

